artemis/reporting/export/deduplication.py

```diff
@@ -0,0 +1,108 @@
+import collections
+import copy
+import datetime
+from typing import Any, DefaultDict, List, Set
+
+from artemis.config import Config
+from artemis.reporting.base.report import Report
+from artemis.reporting.severity import SEVERITY_MAP, Severity
+
+
+def deduplicate_ip_vs_domain_versions(previous_reports: List[Report], reports_to_send: List[Report]) -> List[Report]:
+    """Skips reports if:
+
+    - a vulnerability is on a URL with an IP and an identical report for a domain with this IP has
+      already been sent,
+    - a vulnerability is on a URL with an IP and an identical report for a domain with this IP is
+      planned to be sent,
+    - a vulnerability is on a URL with a domain and an identical report for a URL with an IP of this domain has
+      already been sent.
+    """
```

Comment:
The only thing I'm not a fan of is this function. Very magic, very complicated, very hard to read.